sys/crypto/modes/cbc.c

```diff
@@ -47,23 +54,78 @@ int cipher_encrypt_cbc(const cipher_t *cipher, uint8_t iv[16],
 
         output_block_last = output + offset;
         offset += block_size;
-    } while (offset < length);
+    } while (offset < input_len);
 
     return offset;
 }
 
+int cipher_encrypt_cbc_pkcs7(const cipher_t *cipher, uint8_t iv[16],
+                                    const uint8_t *input, size_t input_len,
+                                    uint8_t *output, size_t output_buffer_size)
```

Comment:
```suggestion
                               const void *input, size_t input_len,
                               void *output, size_t output_buffer_size)
```

makes the API more ergonomic to use

Reply:
Hmm yes, it would make it more versatile.
All of the other cipher APIs use `uint8_t`, though and personally I would find it confusing if there's just one cipher that's different from the rest.

Reply:
eh I'd vote for changing the other ciphers then.

For the user it makes no difference other that they *don't* have to write `(uint8_t *)` when calling the function.

Reply:
Done